### DISCUSSION TITLE
Rootless Docker-Compose Guide

### DISCUSSION THREAD

**Discussion by Maxplosion** [containers/podman#10644] [Q&A] [answered]
Hello, 
I've looked around a bit, but so far was not lucky, so I'll have to try with asking:

Is there some source where I can read about the setup for rootless docker-compose that was introduced in 3.2?

Is it the same as shown in https://www.redhat.com/sysadmin/podman-docker-compose
but without 'sudo' for the docker-compose cmd or is there some other setup required/some other things that one should be aware of when wanting to use rootless docker-compose? 
e.g. do I have to always pass -H unix:/run/user/1000/podman/podman.sock in the compose cmd? Stuff like that is what I would be interested in :)

Thanks in advance!

**Comment by mheon** [comment]
@baude @Luap99 @rhatdan Seems like a good opportunity for a blog?

**Comment by Luap99** [comment, marked as answer]
Yes it is almost the same as described in the blog. Some differences are:
1. use `systemctl --user` instead of `sudo systemctl`
2. test the socket with `curl -H "Content-Type: application/json" --unix-socket $XDG_RUNTIME_DIR/podman/podman.sock http://localhost/_ping`
3. I exported `DOCKER_HOST=unix://$XDG_RUNTIME_DIR/podman/podman.sock`, this allows you to skip the -H on each docker-compose command.
4. you cannot bind ports below 1024 on the host, see https://github.com/containers/podman/blob/master/rootless.md for more shortcomings
5. you cannot directly connect to the container ip from the host, you have to use port forwarding or enter the net namespace with `podman unshare --rootless-cni`
